uml_lib/ebAPI_lib.py

```
@@ -67,58 +71,262 @@
-    return ebTok
+    try:
+        with urlopen(req, timeout=timeout) as resp:
+            charset = resp.headers.get_content_charset() or "utf-8"
+            text = resp.read().decode(charset)
+    except HTTPError as e:
+        # Optionally read error body for debugging
+        try:
+            body = e.read().decode("utf-8", errors="replace")
+        except Exception:
+            body = ""
+        raise HTTPError(e.url, e.code, f"{e.reason}. Response body: {body}", e.headers, e.fp)
```

Ruff (0.12.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)